src/main/java/org/spongepowered/api/event/stats/StatisticChange.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.spongepowered.api.event.stats;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Optional;
+import org.spongepowered.api.entity.player.Player;
+import org.spongepowered.api.stats.Statistic;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents an actual change to a {@link Statistic}.
+ */
+public final class StatisticChange {
+
+    private final Statistic statistic;
+    private final Player player;
+    private final Type type;
+    private final Optional<Long> oldValue;
+    private Optional<Long> newValue;
+
+    /**
+     * Creates a new statistic change for the given player.
+     *
+     * @param statistic The statistic that has changed
+     * @param player The player whose statistic has changed
+     * @param type The type of change
+     * @param oldValue The old value of this statistic
+     * @param newValue The new value for this statistic
+     */
+    public StatisticChange(final Statistic statistic, final Player player, final Type type, @Nullable final Long oldValue, @Nullable final Long newValue) {
+        this.statistic = checkNotNull(statistic, "Statistic cannot be null!");
```

Comment:
As already mentioned, this should be `checkNotNull(statistic, "statistic")`, this will throw a `NullPointerException`, therefore you will know it can't be null.